packages/reports/addon/components/common-actions/schedule.ts

```diff
@@ -134,88 +144,105 @@ export default class ScheduleActionComponent extends Component<Args> {
       return true;
     }
 
-    return !(this.localDeliveryRule?.hasDirtyAttributes && this.isRuleValid);
+    return !(this.localDeliveryRule?.hasDirtyAttributes && this.areRulesValid);
   }
 
   /**
    * @property {Boolean} attemptedSave
    */
   @tracked attemptedSave = false;
 
+  /**
+   * @action addNewRule
+   */
+  @action
+  addNewRule() {
+    this.localDeliveryRule = this._createNewDeliveryRule();
+  }
+
   /**
    * @method _createNewDeliveryRule
    * @private
    * @returns {DS.Model} - new delivery rule model
    */
   _createNewDeliveryRule() {
-    return this.store.createRecord('delivery-rule', {
+    let newRule = this.store.createRecord('delivery-rule', {
       format: { type: this.formats.firstObject },
+      deliveredItem: this.args.model,
+      owner: this.user.getUser(),
     });
+
+    assert('Delivery Rules are defined', this.deliveryRules);
+    this.deliveryRules = [...this.deliveryRules, newRule];
```

Comment:
Can we move this elsewhere? We this seems like a side effect to a method called `_createNewDeliveryRule`

Reply:
line 176